### PR TITLE
fix: ensure that `useSigner` does not broadcast unnecessarily

### PR DESCRIPTION
**Pull request by jxom**
## Description

Addresses https://github.com/wagmi-dev/wagmi/issues/987#issuecomment-1343421969

Seems like `staleTime` doesn't seem to work as intended with `setQueryData`, and can sometimes broadcast to other instances of `useSigner`. Just refactoring our `useSigner` hook to handle invalidaton all in `useQuery`.

## Additional Information

- [x] I read the [contributing docs](/wagmi-dev/wagmi/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

Your ENS/address: jxom.eth

### CODE REVIEW

File: packages/react/src/hooks/accounts/useSigner.ts [hunk 1]
```diff
@@ -28,28 +32,22 @@ export function useSigner<TSigner extends Signer>({
...
 }: UseSignerConfig = {}) {
+  const { connector } = useAccount()
   const chainId = useChainId({ chainId: chainId_ })
   const signerQuery = useQuery<
     FetchSignerResult<TSigner>,
     Error,
     FetchSignerResult<TSigner>,
     ReturnType<typeof queryKey>
-  >(queryKey({ chainId }), queryFn, {
+  >(queryKey({ chainId, connectorId: connector?.id }), queryFn, {
```
Comment (tmm) [inline packages/react/src/hooks/accounts/useSigner.ts:42]: Will `connectorId` be unique enough for when people use multiple Connectors (e.g. connect modal libraries)? We only allow one "active" Connector right now in the wagmi Client, but maybe in the future we can allow multiple - does that break this assumption?

File: packages/react/src/hooks/accounts/useSigner.ts [hunk 2]
```diff
@@ -28,28 +32,22 @@ export function useSigner<TSigner extends Signer>({
...
 
-  const queryClient = useQueryClient()
-  React.useEffect(() => {
-    const unwatch = watchSigner({ chainId }, (signer) =>
-      queryClient.setQueryData(queryKey({ chainId }), signer),
-    )
-    return unwatch
-  }, [queryClient, chainId])
-
   return signerQuery
```
Comment (tmm) [inline packages/react/src/hooks/accounts/useSigner.ts:52]: Can we move the `return` up to the same line as `useQuery(…)` now?

File: packages/react/src/hooks/contracts/usePrepareContractWrite.test.ts [hunk 3]
```diff
@@ -324,6 +324,7 @@ describe('usePrepareContractWrite', () => {
       config[property as keyof typeof config] =
         baseConfig[property as keyof typeof baseConfig]
       rerender()
+      await actConnect({ utils })
```
Comment (tmm) [inline packages/react/src/hooks/contracts/usePrepareContractWrite.test.ts:327]: Interesting. Any idea why we need to call `actConnect` again now? Shouldn't the Client already be connected?

File: packages/react/src/hooks/accounts/useSigner.ts (hunk 1 above)
Comment (jxom) [reply packages/react/src/hooks/accounts/useSigner.ts:42]: True. Might be best if we can make this more unique. Will fix!

File: packages/react/src/hooks/contracts/usePrepareContractWrite.test.ts (hunk 3 above)
Comment (jxom) [reply packages/react/src/hooks/contracts/usePrepareContractWrite.test.ts:327]: Yeah, it's a weird one. The client is already connected, and I even tested this E2E, but not sure why it isn't working here. Might have another look.

File: packages/core/src/actions/accounts/getAccount.ts [hunk 4]
```diff
@@ -67,8 +67,8 @@ export function getAccount<
       } as const
     case 'connecting':
       return {
-        address: undefined,
-        connector: undefined,
+        address: data?.account,
```
Comment (jxom) [inline packages/core/src/actions/accounts/getAccount.ts:70]: Noticed another weird behavior where the account & connector would be `undefined` if you are connecting to another connector.

Current behavior: If you are connected via Injected, then request connection to WalletConnect, the account + connector would be `undefined` (`connecting` status).
Expected behavior: If you are connected via Injected, then request connection to WalletConnect, the account + connector shouldn't change until you are either connected to the requested connector, or disconnected

Comment (tmm) [reply packages/core/src/actions/accounts/getAccount.ts:70]: Makes sense. What is `connector` (below on line 71) set to in this case? For example, I connect with MM and now request a connection to WC: Is `address`, the currently connected MM address and `connector`, the pending WC connector?

**Review by tmm — APPROVED** [review]

**Pull request merged**